controllers/homework.js

```diff
@@ -286,22 +298,26 @@ router.all('/', function (req, res, next) {
                 $populate: ['studentId', 'homeworkId']
             });
             Promise.resolve(submissionPromise).then(submissions => {
-                if(assignment.teacherId === res.locals.currentUser._id){  //teacher
-                    let submissionLength = submissions.filter(function(n){return n.comment != undefined && n.comment != ""}).length;
+                if (assignment.teacherId === res.locals.currentUser._id) {  //teacher
+                    let submissionLength = submissions.filter(function (n) {
+                        return n.comment != undefined && n.comment != "";
+                    }).length;
                     assignment.submissionStats = submissionLength + "/" + assignment.userIds.length;
-                    assignment.submissionStatsPerc = Math.round((submissionLength/assignment.userIds.length)*100);
+                    assignment.submissionStatsPerc = (assignment.userIds.length) ? Math.round((submissionLength / assignment.userIds.length) * 100) : 0;
                     let submissionCount = (submissions.filter(function (a) {
                         return (a.gradeComment != '' || a.grade != null);
                     })).length;
 
                     assignment.gradedStats = submissionCount + "/" + assignment.userIds.length;               // Anzahl der Abgaben
-                    assignment.gradedStatsPerc = Math.round((submissionCount/assignment.userIds.length)*100); // -||- in Prozent
+                    assignment.gradedStatsPerc = (assignment.userIds.length) ? Math.round((submissionCount / assignment.userIds.length) * 100) : 0; // -||- in Prozent
 
                     assignment.averageRating = getAverageRating(submissions, assignment.courseId.gradeSystem);
 
-                }else{ //student
-                    const submission = submissions.filter(function (n) {return n.studentId._id == res.locals.currentUser._id;})[0];  // Abgabe des Schuelers heraussuchen
-                    if (submission != null && submission.comment != ""){ // Abgabe vorhanden?
+                } else { //student
+                    const submission = submissions.filter(function (n) {
+                        return n.studentId._id == res.locals.currentUser._id;
+                    })[0];  // Abgabe des Schuelers heraussuchen
+                    if (submission != null && submission.comment != "") { // Abgabe vorhanden?
```

Comment:
nein, da ein Lehrer auch "keine Abgabe" bewerten kann und dann in submissions.grade oder submissions.gradeComment ein Eintrag steht und somit submissions definiert ist, obwohl der Schüler keine Abgabe getätigt hat.